integration_tests/scheduled_cork_test.go

```diff
@@ -0,0 +1,159 @@
+			s.Require().NoError(err)
+
+			voteMsg := govtypes.NewMsgVote(val.keyInfo.GetAddress(), 1, govtypes.OptionYes)
+			voteResponse, err := s.chain.sendMsgs(*clientCtx, voteMsg)
+			s.Require().NoError(err)
+			s.Require().Zero(voteResponse.Code, "vote error: %s", voteResponse.RawLog)
+		}
+
+		s.T().Log("wait for proposal to be approved")
+		s.Require().Eventuallyf(func() bool {
+			proposalQueryResponse, err := govQueryClient.Proposal(context.Background(), &govtypes.QueryProposalRequest{ProposalId: 1})
+			s.Require().NoError(err)
+			return govtypes.StatusPassed == proposalQueryResponse.Proposal.Status
+		}, time.Second*30, time.Second*5, "proposal was never accepted")
+
+		s.T().Log("verify that contract exists in allowed addresses")
+		val := s.chain.validators[0]
+		s.Require().Eventuallyf(func() bool {
+			kb, err := val.keyring()
+			s.Require().NoError(err)
+			clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kb, "val", val.keyInfo.GetAddress())
+			s.Require().NoError(err)
+
+			queryClient := types.NewQueryClient(clientCtx)
+			res, err := queryClient.QueryCellarIDs(context.Background(), &types.QueryCellarIDsRequest{})
+			if err != nil {
+				s.T().Logf("error: %s", err)
+				return false
+			}
+
+			found := false
+			for _, id := range res.CellarIds {
+				s.T().Logf("managed addresses: %v", res.CellarIds)
+				if common.HexToAddress(id) == counterContract {
+					found = true
+					break
+				}
+			}
+
+			return found
+		}, 10*time.Second, 2*time.Second, "did not find address in managed cellars")
+
+		s.T().Log("schedule a cork for the future")
+		node, err := clientCtx.GetNode()
+		s.Require().NoError(err)
+		status, err := node.Status(context.Background())
+		s.Require().NoError(err)
+		currentBlockHeight := status.SyncInfo.LatestBlockHeight
+
+		s.T().Log("scheduling cork calls")
+		for i, orch := range s.chain.orchestrators {
+			i := i
+			orch := orch
+			s.Require().Eventuallyf(func() bool {
+				clientCtx, err := s.chain.clientContext("tcp://localhost:26657", orch.keyring, "orch", orch.keyInfo.GetAddress())
+				s.Require().NoError(err)
+
+				corkMsg, err := types.NewMsgScheduleCorkRequest(
+					ABIEncodedInc(),
+					counterContract,
+					uint64(currentBlockHeight+5),
+					orch.keyInfo.GetAddress())
+				s.Require().NoError(err, "unable to create cork schedule msg")
+
+				response, err := s.chain.sendMsgs(*clientCtx, corkMsg)
+				if err != nil {
+					s.T().Logf("error: %s", err)
+					return false
+				}
+				if response.Code != 0 {
+					if response.Code != 32 {
+						s.T().Log(response)
+					}
+					return false
+				}
+
+				s.T().Logf("cork msg for %d node sent successfully", i)
+				return true
+			}, 10*time.Second, 500*time.Millisecond, "unable to deploy cork schedule msg for node %d", i)
+		}
+
+		s.T().Logf("checking for updated count in contract")
+		s.Require().Eventuallyf(func() bool {
```

Comment:
I'm not sure this condition fully tests what it is meant to. It wants to see the cellar updated but doesn't inspect what block height it is occurring at. This test would pass if it were processed as a normal cork as well.

Reply:
Sorry, I thought this had been resolved. The previous check basically checks to make sure that the scheduled cork still exists at a block that comes later than the block it was submitted. Right now the only two ways corks are consumed is immediately and at set block height. It's possible that it could be stored and then consumed at an arbitrary height, but I don't know how to test for that